lua/starfall/libs_sh/bit.lua

```diff
@@ -634,13 +634,27 @@ function bit_library.compress(s)
 	return ret
 end
 
---- Decompresses a string
+--- Decompresses a string using LZMA.
 -- @param string s Compressed string to decode
 -- @return string Decompressed string or nil if the input was invalid
-function bit_library.decompress(s)
+function bit_library.decompress(s, maxSize)
 	checkluatype(s, TYPE_STRING)
+	if maxSize ~= nil then
+		checkluatype(maxSize, TYPE_NUMBER)
+		if maxSize > 1e8 then
+			SF.Throw("specified maximum size is too large")
+		end
+	else
+		maxSize = 1e8
+	end
 	if #s > 1e8 then SF.Throw("String is too long!") end
-	local ret = util.Decompress(s)
+	if #s <= 13 then return nil end -- Size of header is 13 bytes, so it can't possibly have any data if it's that size or smaller.
+	if string.sub(s, 6, 13) == '\xff\xff\xff\xff\xff\xff\xff\xff' then
+		-- "streamed" means the uncompressed size isn't specified in the header.
+		-- XZ Utils will always produce streamed LZMA. Use the older LZMA Utils or util.Compress if you need non-streamed.
+		SF.Throw("streamed LZMA is not supported")
```

Comment:
I feel like this should maybe be `return nil, "streamed LZMA is not supported"`, since users might not be expecting this function to error considering the usual behavior is to return `nil`.

Reply:
I would prefer it simply be `return` (nil), and just explain this case in function's description instead.
Also, do not return multiple values. That has a potential to break people's code, if you start returning more than one value; it would be a breaking change, since the function only returned a single value up to *then*.
Consider someone does this in their SF:
```lua
function func(str, tbl)
  local data, key = bit.decompress(tbl.input), tbl[str]
  return key .. data
end
```
Do you see what I mean? Let's say that `decompress` fails due to "streamed LZMA is not supported", the value of `key` would then hold that error string instead of what the user originally intended to store into `key`, that is `tbl[str]`, which would end up discarded if you were to do `return nil, "streamed LZMA is not supported"`.
So, don't.

Reply:
That's not how return values work in Lua.
```lua
--@server
local function a()
    return 'a1', 'a2'
end
local A, B = a(), 'b'
print(A, B) -- a1 b
```

Reply:
It is always a good thing to `pcall` (or use `try`/`catch`) on `bit.decompress`/`bit.compress`, anyways, which will allow the user to catch the error message of `SF.Throw`.

Reply:
I think the behavior should be consistent for all of the ways this function could fail. It would be inconsistent if it threw an error for being streamed but not for other failures.

Reply:
Should it also return `nil` if `maxSize` is too large?

Reply:
> That's not how return values work in Lua.

Ah, yeah you're right, I overlooked this.

Reply:
I prefer the throw since this is such an oddball case. Throwing in all bad cases should've been how this function was designed in the first place though.

Reply:
nil is sufficient as a throw too, however